openssl/src/pkey_ctx.rs

```diff
@@ -0,0 +1,384 @@
+//! PKeyContext operations
+use crate::error::ErrorStack;
+#[cfg(ossl110)]
+use crate::hash::MessageDigest;
+use crate::pkey::{HasPrivate, HasPublic, PKeyRef};
+use crate::{cvt, cvt_p};
+use core::ptr;
+use foreign_types::ForeignTypeRef;
+use std::marker::PhantomData;
+#[cfg(ossl110)]
+use std::os::raw::{c_int, c_uchar};
+
+/// HKDF modes of operation. See [`hkdf_mode`](PKeyContext::hkdf_mode)
+#[cfg(ossl111)]
+pub enum HkdfMode {
```

Comment:
This should be a newtype around an integer with associated constants rather than an enum. See bn::MsbOption as an example.